tgui/packages/tgui/interfaces/common/sorting/SortableTable/index.tsx

```diff
@@ -0,0 +1,163 @@
+  setSortBy: (index: number) => void
+}
+
+const SortableTableHeaderRow = (props: SortableTableHeaderRowProps) => {
+  const { config, sortState, setSortBy } = props;
+
+  return (
+    <Table.Row header>
+      {config.map((config, index: number) => {
+        const { children, sortable, toolTipContent } = config;
+        const needsHeader = sortable === undefined ? true : sortable;
+
+        const sortDirection = !sortState
+          ? null
+          : index === sortState.index
+            ? sortState.dir
+            : null;
+        const sortByIndex = () => setSortBy(index);
+
+
+        if (toolTipContent) {
+          if (needsHeader) {
+
+            return (
+              <Tooltip content={toolTipContent}>
+                <Table.Cell header key={index}>
+                  <Header sortDirection={sortDirection} onSortClick={sortByIndex}>{children}</Header>
+                </Table.Cell>
+              </Tooltip>);
+          }
+          else {
+            return (
+              <Tooltip content={toolTipContent}>
+                <Table.Cell header key={index}>
+                  {children}
+                </Table.Cell>
+              </Tooltip>);
+          }
+        } else {
+
+          if (needsHeader) {
+            return (
+              <Table.Cell header key={index}>
+                <Header sortDirection={sortDirection} onSortClick={sortByIndex}>{children}</Header>
+              </Table.Cell>);
+          } else {
+            return (
+              <Table.Cell header key={index}>
+                {children}
+              </Table.Cell>);
+
+          }
+        }
+      })}
+    </Table.Row>
+  );
+};
+
+const SortableTableRow = (props: SortableTableRowProps) => {
+  const { cells, ...rest } = props;
+  return (
+
+    <Table.Row
+      {...rest}>
+      {cells.map((config, index) => {
+        const { children, wrapInCell, ...rest } = config;
+        if (!children) {
+          return;
+        }
+        let wrapCell = wrapInCell === undefined
+          ? true
+          : wrapInCell;
+
+        if (wrapCell) {
+
+          return (
+            <Table.Cell
+              key={index}
+              {...rest}>
+              {children}
+            </Table.Cell>);
+        } else {
+          return (
+            <Fragment
+              key={index}
+              {...rest}>
+              {children}
+            </Fragment>);
+        }
+      })}
+
+    </Table.Row>);
+};
+
+interface SortableTableProps {
+  /** Configure the header row of the table */
+  headerConfig: SortableTableHeaderConfig[]
+
+  /** Row data */
+  rowData: SortableTableRowProps[],
+
+  /** A name for the table for use with useLocalState / useSharedState */
+  name: string,
+
+  /** Use local state or shared state; Default: true */
+  useLocalState?: BooleanLike,
+
+  /** Data required for searching through rows */
+  searchState?: SearchState
+
+  /** Props, if provided to be spread on <Table> element */
+  [x: keyof any]: unknown
```

Comment:
Better to define this as a specific prop, to avoid accidental inclusion.

i.e. `tableProps?: TableProps`, then earlier on `type TableProps = Record<string, unknown>` (or whatever you need to do to make TypeScript appeased)